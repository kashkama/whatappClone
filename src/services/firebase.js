import * as firebase from "firebase";

export const initialize = () => firebase.initializeApp({
    apiKey: process.env.apiKey,
    authDomain: "whatsappclone-20181.firebaseapp.com",
    databaseURL: "https://whatsappclone-20181.firebaseio.com",
    projectId: "whatsappclone-20181",
    storageBucket: "whatsappclone-20181.appspot.com",
    messagingSenderId: "315651881181"
});

export const setListener = (endpoint, updaterFn) => {
    firebase.database().ref(endpoint).on("value", updaterFn);
    return () => firebase.database().ref(endpoint).off();
}

export const pushData = (endpoint, data) => {
    return firebase.database().ref(endpoint).push(data);
}