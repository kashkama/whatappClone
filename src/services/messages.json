{
    "messages" :[
        {
            "incoming": true,
            "message": "Hi Owen"
        },
        {
            "incoming": false,
            "message": "Hi, Vader"
        },
        {
            "incoming": true,
            "message": "When we will learn real data fetching?"
        },
        {
            "incoming": false,
            "message": "First we need to learn debugging and testing, then we will learn styling and animations and then we will learn real data fetching. Let's use this mock data for now ok?"
        }
    ]
}