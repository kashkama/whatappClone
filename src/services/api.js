import { initialize, setListener, pushData } from "./firebase";

const mockMessages = [
    {
        incoming: true,
        message: "Hi Owen"
    },
    {
        incoming: false,
        message: "Hi, Vader"
    },
    {
        incoming: true,
        message: "When we will learn real data fetching?"
    },
    {
        incoming: false,
        message: "First we need to learn debugging and testing, then we will learn styling and animations and then we will learn real data fetching. Let's use this mock data for now ok?"
    }
];

export const getMockData = () => (
    new Promise(resolve => setTimeout(() => resolve(mockMessages), 1000))
)

export const initApi = () => initialize();

export const getMessages = (updaterFn) => setListener("messages", updaterFn);

export const postMessage = (message) => {
    if(Boolean(message)) {
        pushData("messages", {
            incoming: false,
            message
        })
    }
};