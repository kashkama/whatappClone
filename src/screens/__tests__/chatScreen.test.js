import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { shallow, configure} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { getMockDataSpy } from "./../../services/api";
import ChatScreen from "./../chatScreen";

configure({ adapter: new Adapter()});
jest.mock("../../services/api");

it("renders ChatScreen component correctly", () => {
    expect(ChatScreen).toMatchSnapshot();
})

it("ChatScreen has proper navigation options", () => {
    expect(ChatScreen.navigationOptions({
        navigation: {
            state: {
                params: {
                    name: "john"
                }
            }
        }
    })).toMatchSnapshot();
})

// it("navigate to home on press", () => {
//     const goBack = jest.fn();
//     const chatScreen = shallow(
//         <ChatScreen navigation={{ goBack }}/>
//     )
//     chatScreen.find("Button").simulate("press");
//     expect(goBack).toBeCalledWith();
// });

it("fetches data when component is rendered", () => {
    const chatScreen = renderer.create(
        <ChatScreen/>
    );
    expect(getMockDataSpy).toBeCalled();
});