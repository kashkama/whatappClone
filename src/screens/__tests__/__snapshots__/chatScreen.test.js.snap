// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatScreen has proper navigation options 1`] = `
Object {
  "title": "Chat with john",
}
`;

exports[`renders ChatScreen component correctly 1`] = `[Function]`;
